### PR TITLE
Refactor validation dispatchers

### PR DESCRIPTION
**Pull request by gbirke**
Wrap each validator in a validator-specific `ValidationDispatcher` instance.

### CODE REVIEW

File: app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+		testStore = { dispatch: sinon.stub() },
+		dispatcher = createAddressValidationDispatcher(
+			validator,
+			initialData
+		);
+
+	dispatcher.dispatchIfChanged( testData, testStore );
+
+	t.ok( validator.validate.calledOnce, 'validation function is called once' );
+	t.ok( validator.validate.calledWith( expectedData ), 'validation function is called with selected fields' );
```
Comment (JeroenDeDauw) [inline app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js]: I'm not a fan of having a var for everything and putting those all at the start of the scope. That results in more state, mutability and distance between related stuff. For instance for this `expectedData`, if you inline it, you have none of those downsides.

File: app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+		validator = { validate: sinon.stub().returns( successResult ) },
+		testStore = { dispatch: sinon.spy() },
+		dispatcher = createAddressValidationDispatcher(
+			validator,
+			initialData
+		);
+
+	dispatcher.dispatchIfChanged( testData, testStore );
+
+	t.ok( testStore.dispatch.calledWith( Actions.newFinishAddressValidationAction( successResult ) ) );
```
Comment (JeroenDeDauw) [inline app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js]: I was expecting the whole value to be specified in the test here. Won't that also catch errors in `newFinishAddressValidationAction`? Based on our conversation I figured this is not tested elsewhere. (I'm not sure this should be done here or would actually be helpful if it was)

Comment (gbirke) [inline app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js]: I'd agree with you in other cases. In this case, actions are basically value objects and action creation functions like `newFinishAddressValidationAction` are constructors of value objects and those aren't tested, right?

File: app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js (hunk 1 above)
Comment (gbirke) [inline app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js]: I agree for `expectedData` and have put up PR #806

File: app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js (hunk 2 above)
Comment (JeroenDeDauw) [inline app/js/tests/validation_dispatchers/test_address_validation_dispatcher.js]: I was thinking about catching mismatches between which fields those value objects have and what the code using them expects.

**Pull request merged**